ksql-engine/src/main/java/io/confluent/ksql/function/udaf/sum/DecimalSumKudaf.java

```diff
@@ -26,34 +26,37 @@
 import org.apache.kafka.connect.data.Schema;
 import org.apache.kafka.connect.data.Struct;
 import org.apache.kafka.streams.kstream.Merger;
+import org.apache.kafka.streams.kstream.ValueMapper;
 
 public class DecimalSumKudaf
-    extends BaseAggregateFunction<BigDecimal, BigDecimal>
-    implements TableAggregationFunction<BigDecimal, BigDecimal> {
+    extends BaseAggregateFunction<BigDecimal, BigDecimal, BigDecimal>
+    implements TableAggregationFunction<BigDecimal, BigDecimal, BigDecimal> {
 
   private final MathContext context;
 
   DecimalSumKudaf(
       final String functionName,
       final int argIndexInValue,
-      final Schema returnSchema
+      final Schema aggregateSchema,
```

Comment:
I don't think you need both aggregateSchema and outputSchema in your constructor. They are always the same.